zero-carbon-bill/combine.py

```diff
@@ -0,0 +1,154 @@
+import json
+from glob import glob
+import csv
+
+def sort_by_line_number(elem):
+    return elem['line_number']
+
+def sort_by_weight(elem):
+    return elem['weight']
+
+input_file_pattern = 'input/*.json'
+input_folder_name = 'input/'
+output_folder_name = 'combined/'
+
+filenames = glob(input_file_pattern)
+
+# Import key phrases
+key_phrases_file = 'key-phrases/key-phrases.json'
+with open(key_phrases_file) as f:
+    contents = f.read()
+
+raw_key_phrases = json.loads(contents)
+key_phrases = {}
+
+for item in raw_key_phrases:
+    filename = item['File']
+    if key_phrases.get(filename, None) is None:
+        key_phrases[filename] = []
+    phrases = item['KeyPhrases']
+    for phrase in phrases:
+        phrase.pop('BeginOffset')
+        phrase.pop('EndOffset')
+    key_phrases[filename] = phrases
+
+# Import sentiment analysis
+
+sentiment_analysis_file = 'sentiment-analysis/output.json'
+with open(sentiment_analysis_file) as f:
+    contents = f.read()
+
+raw_sentiment_analysis = json.loads(contents)
+sentiment_analysis = {}
+
+for item in raw_sentiment_analysis:
+    filename = item['File']
+    if sentiment_analysis.get(filename, None) is None:
+        sentiment_analysis[filename] = []
+    sentiment_analysis[filename].append(item)
+
+# Import topic modelling: topics
+
+topic_terms = {}
+
+with open('topic-modelling/topic-terms.csv', newline='') as topic_terms_file:
+    reader = csv.reader(topic_terms_file, delimiter=',')
+    for row in reader:
+        if row[0] == 'topic':
+            next
+        else:
+            topic_number = row[0]
+            item = {'term': row[1], 'weight': row[2]}
+            if topic_terms.get(topic_number, None) is None:
+                topic_terms[topic_number] = []
+            topic_terms[topic_number].append(item)
+
+for key, value in topic_terms.items():
+    topic_terms[key] = sorted(value, key=sort_by_weight, reverse=True)
+
+
+# Import topic modelling: documents
+
+docs_with_topics = {}
+
+with open('topic-modelling/doc-topics.csv', newline='') as doc_topics:
```

Comment:
this "topic-modelling/doc-topics.csv" could be in a variable set at the top of the file.

Reply:
done in 999d2ad